code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/ProductCatalogNotificationBuilder.kt

```diff
@@ -0,0 +1,335 @@
+            }
+
+            // create a pending intent for the thumbnail
+            val pendingIntentThumbnailInteraction = createThumbnailInteractionPendingIntent(
+                context,
+                broadcastReceiverClass,
+                channelIdToUse,
+                pushTemplate,
+                index
+            )
+            expandedLayout.setOnClickPendingIntent(
+                thumbIds[index],
+                pendingIntentThumbnailInteraction
+            )
+        }
+    }
+
+    /**
+     * Sets the CTA button for the product catalog notification.
+     *
+     * @param context the application [Context]
+     * @param trackerActivityClass the [Class] of the activity to set in the created pending intent for tracking purposes
+     * @param expandedLayout the [RemoteViews] object containing the expanded layout of the push template notification
+     * @param pushTemplate the [ProductCatalogPushTemplate] object containing the product catalog push template data
+     */
+    private fun setupCtaButton(
+        context: Context,
+        trackerActivityClass: Class<out Activity>?,
+        expandedLayout: RemoteViews,
+        pushTemplate: ProductCatalogPushTemplate
+    ) {
+        // apply the color to the cta button
+        expandedLayout.setCtaButtonColor(R.id.cta_button, pushTemplate.ctaButtonColor)
+
+        // apply text to the cta button
+        expandedLayout.setTextViewText(R.id.cta_button, pushTemplate.ctaButtonText)
+
+        // apply the open uri action to the cta button
+        expandedLayout.setOnClickPendingIntent(
+            R.id.cta_button,
+            PendingIntentUtils.createPendingIntent(
+                context,
+                trackerActivityClass,
+                pushTemplate.ctaButtonUri,
+                PushTemplateConstants.CatalogActionIds.CTA_BUTTON_CLICKED,
+                pushTemplate.tag,
+                pushTemplate.isNotificationSticky ?: false
+            )
+        )
+    }
+
+    /**
+     * Sets custom colors to the product catalog CTA button.
+     *
+     * @param containerViewId [Int] containing the resource id of the push template notification RemoteViews
+     * @param buttonColor [String] containing the hex color code for the cta button
+     */
+    private fun RemoteViews.setCtaButtonColor(
+        containerViewId: Int,
+        buttonColor: String?
+    ) {
+        // get custom color from hex string and set it the cta button
+        if (buttonColor.isNullOrEmpty()) {
+            Log.trace(
+                LOG_TAG,
+                SELF_TAG,
+                "Empty cta button color hex string found, custom color will not be applied to the cta button."
+            )
+            return
+        }
+        setElementColor(
+            containerViewId,
+            "#$buttonColor",
+            PushTemplateConstants.MethodNames.SET_BACKGROUND_COLOR,
+            PushTemplateConstants.FriendlyViewNames.CTA_BUTTON
```

Comment:
just curious what might happen here. if the buttonColor is an invalid hex. Does it gracefully fail at run time?
Wondering if we have to introduce a sanity check for hexString.
- If this needs fix. Not that this has to be solved in this PR. We can take it later and cleanup all the templates

Reply:
you are right, we should sanity check that its a valid color. i'll see how we can best do that.

Reply:
`setElementColor` verifies that the color is valid

Reply:
ah cool, thanks for the reminder